src/test/java/org/minions/devfund/angela/SquarelotronTest.java

```diff
@@ -0,0 +1,163 @@
+package org.minions.devfund.angela;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for {@link Squarelotron} .
+ */
+public class SquarelotronTest {
```

Comment:
It is missing unit tests when the ring is greater than 1.

Please consider the following unit test.
```java
    @Test
    public void testMainDiagonalFlipDimension3Ring2() {
        final int ringNumber = 2;
        final int dimension = 3;
        final Squarelotron squarelotron = new Squarelotron(dimension);
        final int[][] expectedMatrix = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
        Squarelotron result = squarelotron.mainDiagonalFlip(ringNumber);
        assertArrayEquals(expectedMatrix, result.getSquarematrix());
    }
```